common/src/main/java/com/microsoft/identity/common/internal/platform/AndroidDevicePopManager.java

```diff
@@ -205,7 +205,7 @@ private KeyPair generateNewRsaKeyPair(@androidx.annotation.NonNull final Context
                         Logger.error(TAG, "Import unsupported. Skipping import flag then retry.", e);
                         tryImport = false;
 
-                        if (tryStrongBox && null != e.getCause() && isStrongBoxUnavailableException(e.getCause())) {
+                        if (tryStrongBox && null != e.getCause() && (isStrongBoxUnavailableException(e.getCause()) || isNegativeInternalError(e.getCause()))) {
```

Comment:
Is this negative internal error check necessary?

Reply:
Yes, I cannot reproduce the issue now but I saw that when I removed this, i started seeing some errors. I believe there is no harm in keeping it as it is only adding a condition to skip strongbox when an unknown error occurs